packages/snowpack/src/index.js

```diff
@@ -58,14 +29,35 @@ export default function preactRefreshPlugin(config, pluginOptions) {
             };
           };
 
-          window.$RefreshReg$ = (type, id) => {};
+          window.$RefreshReg$ = (type, id) => {
+            __module_exports__.push(type.name);
+          };
 
           ${contents}
 
           window.$RefreshSig$ = prevRefreshSig;
           window.$RefreshReg$ = prevRefreshReg;
 
-          ${hasRegister || hasJsx || isComponent(lastPart) ? postLude : ''}
+          if (__module_exports__.some(shouldPrefreshBind) && import.meta.hot) {
```

Comment:
super nit: `if (import.meta.hot && __module_exports__.some(shouldPrefreshBind))` makes it more likely that the reader will see the "this is HMR-related" piece of the conditional

Reply:
```suggestion
          if (import.meta.hot && __module_exports__.some(shouldPrefreshBind)) {
```

Good suggestion thanks